### ISSUE TITLE
Outdated Guava causing binary incompatibility

### ISSUE THREAD

**Issue by peterfigure** [auth0/jwks-rsa-java#136]
### Describe the problem

```
Exception in thread "main" java.lang.NoSuchMethodError: 'com.google.common.cache.CacheBuilder com.google.common.cache.CacheBuilder.expireAfterWrite(java.time.Duration)'
   at com.auth0.jwk.GuavaCachedJwkProvider.<init>(GuavaCachedJwkProvider.java:54)
   at com.auth0.jwk.JwkProviderBuilder.build(JwkProviderBuilder.java:180)
```

Using the current version: `0.20.0` along with any recent version of the Google SDK (which brings in 31.0.x of Guava).

**Comment by wduqu001** [comment]
> ### Describe the problem
> ```
> Exception in thread "main" java.lang.NoSuchMethodError: 'com.google.common.cache.CacheBuilder com.google.common.cache.CacheBuilder.expireAfterWrite(java.time.Duration)'
>    at com.auth0.jwk.GuavaCachedJwkProvider.<init>(GuavaCachedJwkProvider.java:54)
>    at com.auth0.jwk.JwkProviderBuilder.build(JwkProviderBuilder.java:180)
> ```
> 
> Using the current version: `0.20.0` along with any recent version of the Google SDK (which brings in 31.0.x of Guava).

Perhaps it may be of use to you... I solved this problem by using the UrlJwkProvider instead of the JwkProviderBuilder.

**Comment by jimmyjames** [comment]
Thanks for reporting this; I reproduced this in a test project by including `com.google.api-client:google-api-client:1.31.5` as a dependency. It appears this is bringing in `com.google.guava:guava:30.1.1-android` which is causing this issue. It appears related to the change in #130, I'll look into this further, perhaps including the `com.google.guava:guava:31.0.1-jre` dependency as well could be a workaround?

**Comment by jimmyjames** [comment]
@peterfigure or @wduqu001 I've made a change in #140 to address this. Could be argued that the real bug is in Guava given the Android implementation doesn't support `Duration` (and as discussed in #140 there's a comment that suggests it should), but the change in #140 is straightforward enough and should address your issue. Would either one of you be willing to try it out and let me know if it fixes the issue? Thanks!

**Comment by jimmyjames** [comment]
#140 to fix this issue has been released in version 0.20.2, it should be available in maven central shortly. Thanks!
